lib/models/package-info-cache/package-info.js

```diff
@@ -346,21 +347,26 @@ class PackageInfo {
     if (invalidPackages.length > 0) {
       let typeName = this.project ? 'project' : 'addon';
 
-      logger.info('');
-      logger.info(`The 'package.json' file for the ${typeName} at ${this.realPath}`);
+      let msg = `The 'package.json' file for the ${typeName} at ${this.realPath}`;
 
       let relativePath;
-
       if (invalidPackages.length === 1) {
         relativePath = path.relative(this.realPath, invalidPackages[0].realPath);
-        logger.info(`  specifies an invalid, malformed or missing addon at relative path '${relativePath}'`);
+        msg = `${msg}\n  specifies an invalid, malformed or missing addon at relative path '${relativePath}'`;
       } else {
-        logger.info('  specifies invalid, malformed or missing addons at relative paths');
+        msg = `${msg}\n  specifies invalid, malformed or missing addons at relative paths`;
         invalidPackages.forEach(packageInfo => {
           let relativePath = path.relative(this.realPath, packageInfo.realPath);
-          logger.info(`    '${relativePath}'`);
+          msg = `${msg}\n    '${relativePath}'`;
         });
       }
+
+      const config = require('../../utilities/get-config')();
+      if (config.get('errorOnInvalidAddon')) {
```

Comment:
Why is this a flag and not the default? What circumstances are we intentionally wanting to swallow invalid addon paths for?

Reply:
I know, right? This has bitten us many times, I think the reason for the flag is that the behavior is radically different from the current behavior. Previously it didn't even throw a warning without the DEBUG flag set...

Reply:
So the reason it is behind a flag is that it is a potentially breaking change?

Reply:
Ya, per @stefanpenner

Reply:
I think we should:

* have it always emit a warning to the console (this is the only way to get these cleaned up throughout the ecosystem)
* have it throw an error when a `process.env` variable is set

Reply:
OK, so this PR does the first one already, so I will just switch the flag from .ember-cli to an env...